src/sctools/platform.py

```diff
@@ -6,45 +6,40 @@ class PlatformBase:
     """Abstract base class for platform specific metadata"""
 
     @classmethod
-    def get_tags(cls, sequencing_read):
+    def attach_barcodes(cls, args=None):
+        """command line entrypoint for attaching barcodes to a bamfile."""
         raise NotImplementedError
 
+
+class TenXV2(PlatformBase):
+
+    cell_barcode = fastq.EmbeddedBarcode(start=0, end=16, quality_tag='CY', sequence_tag='CR')
+    molecule_barcode = fastq.EmbeddedBarcode(start=16, end=26, quality_tag='UY', sequence_tag='UR')
+    sample_barcode = fastq.EmbeddedBarcode(start=0, end=8, quality_tag='SY', sequence_tag='SR')
+
     @classmethod
-    def tag_bamfile(cls, input_bamfile_name, output_bamfile_name, files_with_tags):
+    def tag_bamfile(cls, input_bamfile_name, output_bamfile_name, tag_generators):
         """Extracts tags from fastq files_with_tags, attaches them to input_bamfile_name and writes
         the result to output_bamfile_name
 
         :param str input_bamfile_name: input bam
         :param str output_bamfile_name: output bam
-        :param dict files_with_tags: dict mapping {tags: filenames}
+        :param list tag_generators: list of generators of tags
         """
-        tag_generators = []
-        for k, v in files_with_tags.items():
-            tag_generators.append(fastq.EmbeddedBarcodeGenerator(cls.get_tags(k), files=v))
-
         bam_tagger = bam.Tagger(input_bamfile_name)
         bam_tagger.tag(output_bamfile_name, tag_generators)
 
     @classmethod
-    def attach_barcodes(cls, args=None):
-        """command line entrypoint for attaching barcodes to a bamfile."""
-        raise NotImplementedError
-
-
-class TenXV2(PlatformBase):
-
-    cell_barcode = fastq.EmbeddedBarcode(start=0, end=16, quality_tag='CY', sequence_tag='CR')
-    molecule_barcode = fastq.EmbeddedBarcode(start=16, end=24, quality_tag='UY', sequence_tag='UR')
-    sample_barcode = fastq.EmbeddedBarcode(start=0, end=8, quality_tag='SY', sequence_tag='SR')
-
-    _tags = {
-        'r1': (cell_barcode, molecule_barcode),
-        'i1': (sample_barcode,)
-    }
-
-    @classmethod
-    def get_tags(cls, sequencing_read):
-        return cls._tags[sequencing_read]
+    def make_tag_generators(cls, r1, i1, whitelist=None):
```

Comment:
Can you add a docstring describing what this method is doing and what the expected inputs/outputs are?